packages/datadog-plugin-google-cloud-pubsub/src/index.js

```diff
@@ -40,23 +40,65 @@ function createWrapRequest (tracer, config) {
 function createWrapSubscriptionEmit (tracer, config) {
   return function wrapSubscriptionEmit (emit) {
     return function emitWithTrace (eventName, message) {
-      if (eventName !== 'message' || !message) {
+      if (eventName !== 'message' || !message || !message._datadog_span) {
         return emit.apply(this, arguments)
       }
 
-      const topic = this.metadata && this.metadata.topic
+      const span = message._datadog_span
+
+      return tracer.scope().activate(span, () => {
+        try {
```

Comment:
Why would emit throw?

Reply:
If one of the listener throws, we need to capture that error. However, it should re-throw which is not the case right now, so I'll fix that.